app/questionnaire/questionnaire_schema.py

```diff
@@ -887,21 +968,119 @@ def _get_error_messages(self) -> dict:
+
+            if rules_section_dependencies:
                 self._when_rules_section_dependencies_by_section[
                     section["id"]
                 ] = rules_section_dependencies
 
+            for (
+                key,
+                values,
+            ) in rule_section_dependencies_for_progress_value_source.items():
+                progress_section_dependencies[key].update(values)
+
+            self._populate_block_dependencies_for_progress_value_source(
+                rule_block_dependencies_for_progress_value_source
+            )
+
+    def _populate_block_dependencies_for_progress_value_source(
+        self,
+        rule_block_dependencies_for_progress_value_source: dict[
+            str, DependencyDictType
+        ],
+    ) -> None:
+        """
+        Populates section dependencies for progress value sources at the block level
+        """
+        dependencies = (
+            self._when_rules_block_dependencies_by_section_for_progress_value_source
+        )
+        for (
+            dependent_section,
+            section_dependencies_by_block,
+        ) in rule_block_dependencies_for_progress_value_source.items():
+            for block_id, section_ids in section_dependencies_by_block.items():
+                dependencies[dependent_section][block_id].update(section_ids)
+
+    def _get_section_and_block_ids_dependencies_for_progress_source_and_answer_ids_from_rule(
+        self, current_section_id: str, rule: Mapping
+    ) -> tuple[set[str], dict[str, dict[str, OrderedSet[str] | DependencyDictType]]]:
+        """
+        For a given rule, returns a set of dependent answer ids and any dependent sections for progress value sources.
+        Progress dependencies are keyed both by section and by block e.g.
+        sections: {"section-1": {"section-2"}}
+        blocks: {"section-1": {"block-1": {"section-2"}}}
+        """
+        answer_id_list: set[str] = set()
+        dependencies_ids_for_progress_value_source: dict[
+            str, dict[str, OrderedSet[str] | DependencyDictType]
+        ] = {
+            "sections": {},
+            "blocks": {},
+        }
+        identifier: str | None = rule.get("identifier")
+        source: str | None = rule.get("source")
+        selector: str | None = rule.get("selector")
+
+        if source == "answers" and identifier:
+            answer_id_list.add(identifier)
+        elif source == "calculated_summary" and identifier:
+            calculated_summary_block = self.get_block(identifier)
+            # Type Ignore: Calculated summary block will exist at this point
+            calculated_summary_answer_ids = get_calculated_summary_answer_ids(
+                calculated_summary_block  # type: ignore
+            )
+            answer_id_list.update(calculated_summary_answer_ids)
+        elif source == "progress" and identifier and selector:
+            if selector == "section":
```

Comment:
These two ifs could be combined

Reply:
Removed `selector` from the first line as it wasn't needed

Reply:
was thinking you could also do `if selector == "section" and identifier != current_section_id`